src/polycubed/src/controller.cpp

```diff
@@ -445,7 +447,18 @@ void Controller::send_packet_to_cube(uint16_t module_index, uint16_t port_id,
   }
   metadata_table_->update_value(ctrl_rx_md_index_, md_temp);
 
-  iface_->send(const_cast<std::vector<uint8_t> &>(packet));
+  if (sense == service::Sense::EGRESS) {
+      iface_->send(const_cast<std::vector<uint8_t> &>(packet));
+  } else {
+      /* ingress packet needs modifying the destination MAC address
```

Comment:
I don't totally agree with this change as it will limit the packets that a transparent cube is able to inject in the fast path. I think the best approach is not to change it and let it under the responsibility of the user.

Reply:
A packet received from a physical NIC has destination MAC to this NIC, when it was forwarded to the slow path and injected back to pcn_tc_cp interface, the packet will be dropped by Linux kernel because the packet doesn't have a valid MAC address for IP processing. In my testing the packet will be classified as PACKET_OTHERS and dropped before calling the ip_rcv().

Reply:
I think this is related to my previous comment, the OS discards the packet because it appears to be coming from the `pcn_tc_cp` interface.  If there is not another solution for this problem I'll ask to scope this modification only to the case a transparent service is attached to a NIC.

Reply:
Sure this is a good point I think we should make this change to transparent scope. Thanks for pointing out this.